ledger/participant-integration-api/src/main/scala/platform/store/backend/oracle/OracleField.scala

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.oracle
+
+import java.lang
+import java.sql.{PreparedStatement, Timestamp}
+import java.time.Instant
+import com.daml.platform.store.backend.common.{Field, TrivialField, TrivialOptionalField}
+
+private[oracle] case class OracleStringField[FROM](extract: FROM => String)
+    extends TrivialField[FROM, String]
+
+private[oracle] case class OracleStringOptional[FROM](extract: FROM => Option[String])
+    extends TrivialOptionalField[FROM, String]
+
+private[oracle] case class OracleBytea[FROM](extract: FROM => Array[Byte])
+    extends TrivialField[FROM, Array[Byte]]
+
+private[oracle] case class OracleByteaOptional[FROM](extract: FROM => Option[Array[Byte]])
+    extends TrivialOptionalField[FROM, Array[Byte]]
+
+private[oracle] case class OracleIntOptional[FROM](extract: FROM => Option[Int])
+    extends Field[FROM, Option[Int], java.lang.Integer] {
+  override def convert: Option[Int] => Integer = _.map(Int.box).orNull
+  override def prepareDataTemplate(
+      preparedStatement: PreparedStatement,
+      index: Int,
+      value: java.lang.Integer,
+  ): Unit = {
+    // TODO FIXME either cleanup and use Field from common, or change the line below
+    preparedStatement.setObject(index, value)
+  }
+}
+
+private[oracle] case class OracleBigint[FROM](extract: FROM => Long)
+    extends TrivialField[FROM, Long] {
+  override def prepareDataTemplate(
+      preparedStatement: PreparedStatement,
+      index: Int,
+      value: Long,
+  ): Unit = {
+    // TODO FIXME either cleanup and use Field from common, or change the line below
+    preparedStatement.setObject(index, value)
+  }
+}
+
+private[oracle] case class OracleSmallintOptional[FROM](extract: FROM => Option[Int])
+    extends Field[FROM, Option[Int], java.lang.Integer] {
+  override def convert: Option[Int] => Integer = _.map(Int.box).orNull
+  override def prepareDataTemplate(
+      preparedStatement: PreparedStatement,
+      index: Int,
+      value: java.lang.Integer,
+  ): Unit = {
+    // TODO FIXME either cleanup and use Field from common, or change the line below
+    preparedStatement.setObject(index, value)
+  }
+}
+
+private[oracle] case class OracleBooleanField[FROM](extract: FROM => Boolean)
+    extends TrivialField[FROM, Boolean]
+//  {
```

Comment:
remove?

Reply:
yeah still need to clean up unused fields here